### ISSUE TITLE
Could not resolve host: downloads.hypriot.com

### ISSUE THREAD

**Issue by Khoa-NT** [romilly/rpi-docker-tensorflow#1]
Hi,

I followed step 1 Install Docker on your Raspberry Pi to install tensorflow on my **Pi Zero** and got this error 

**pi@raspberrypi:~ $ curl -sSL http://downloads.hypriot.com/docker-hypriot_1.10.3-1_armhf.deb >/tmp/docker-hypriot_1.10.3-1_armhf.deb
curl: (6) Could not resolve host: downloads.hypriot.com**

Do you have any solutions.

Thank you

**Comment by samjabrahams** [comment]
Looks like they moved the file- see if this line works instead:

```
curl -sSL https://packagecloud.io/Hypriot/Schatzkiste/packages/debian/wheezy/docker-hypriot_1.10.3-1_armhf.deb/download >/tmp/docker-hypriot_1.10.3-1_armhf.deb
```

**Comment by Khoa-NT** [comment]
Thank you, hope romilly update the link

I got error in the last step:

pi@raspberrypi:~/tensorflow/rpi-docker-tensorflow/build-tensor-pi $ **docker build -t='khoa/rpi-docker-tensorflow' .
Cannot connect to the Docker daemon. Is the docker daemon running on this host?**

or 
pi@raspberrypi:~/tensorflow/rpi-docker-tensorflow/build-tensor-pi $ **sudo docker build -t='khoa/rpi-docker-tensorflow' .
Cannot connect to the Docker daemon. Is the docker daemon running on this host?**

I check `docker version` :
pi@raspberrypi:~/tensorflow/rpi-docker-tensorflow/build-tensor-pi $ **docker version
Client:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.4.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 22:23:48 2016
 OS/Arch:      linux/arm
Cannot connect to the Docker daemon. Is the docker daemon running on this host?**

**Comment by samjabrahams** [comment]
Correct step is below.

**Comment by Khoa-NT** [comment]
I found **decomads**'s solution here : 
https://forums.docker.com/t/cannot-connect-to-the-docker-daemon-is-the-docker-daemon-running-on-this-host/8925/5

`sudo service docker stop`

`sudo service docker start`

then redo the step
`sudo docker build -t='khoa/rpi-docker-tensorflow' .`

Now the terminal is printing 
*_Sending build context to Docker daemon  31.7 MB
Step 1 : FROM resin/rpi-raspbian
latest: Pulling from resin/rpi-raspbian
395823d8c49b: Extracting  44.5 MB/45.86 MB
f79b3906acf0: Download complete 
2e4eaf000ed0: Download complete 
44f82080e2cc: Download complete 
a3ed95caeb02: Download complete 
f23aeb340745: Download complete 
466adec6a1f2: Download complete 
281ed5189bce: Download complete 
95c0246ab315: Download complete *_ 

is it right ?

**Comment by samjabrahams** [comment]
Looks like a good start. We won't know if it's working until you try to spin up a container and use it.

**Comment by Khoa-NT** [comment]
haiz, after waiting it finish, I got the same error in this topic. 
 https://github.com/samjabrahams/tensorflow-on-raspberry-pi/issues/43

```
Step 6 : ADD tensorflow-0.9.0-cp27-none-linux_armv7l.whl .
 ---> 6f4ee4d635c7
Removing intermediate container 91f9078cbf1f
Step 7 : RUN pip install tensorflow-0.9.0-cp27-none-linux_armv7l.whl
 ---> Running in 0a207d9c2150
tensorflow-0.9.0-cp27-none-linux_armv7l.whl is not a supported wheel on this platform.
The command '/bin/sh -c pip install tensorflow-0.9.0-cp27-none-linux_armv7l.whl' returned a non-zero code: 1
```

Feel hopeless with Pi zero, it seems tensorflow cannot work on the ARMV6 

I already tried this tutorial and stuck too
https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/pi_examples
Still waiting for the solution in this post:
https://github.com/tensorflow/tensorflow/issues/3251#issuecomment-256768778

Would you mind helping me.

1/ Are there any ways to install tensorflow with C++ or QT on Pi ? Cause I can use QT on Pi zero. 

2/ If I can not install tensorflow on pi zero, are there any ways  to use/read the file result **graph.pb** ( trainning by tensorflow on ubuntu) without install tensorflow.
My ideal is trainning the tensorflow project on ubuntu, then copy file result **graph.pb** into pi zero. Pi zero will use QT, python or C to read the **graph.pb** . 
If I input the picture, Pi zero will run **graph.pb** and return the result of recognition.

**Comment by romilly** [comment]
I don't think the image will run (well enough to be useful) on a Pi zero because the zero only has 512MB of Ram. It's only just usable on the RPi model 3B.

It looks as if Hypriot have removed their deb package because Docker now officially support ARM architecture. I will investigate and update the instructions.

**Comment by Khoa-NT** [comment]
So back to the question 
2/ If I can not install tensorflow on pi zero, are there any ways to use/read the file result **graph.pb** ( trainning by tensorflow on ubuntu).
My ideal is trainning the tensorflow project on ubuntu, then copy file result **graph.pb** into Pi zero. Pi zero will use QT, python or C to read the graph.pb .
If I input the picture, Pi zero will run graph.pb and return the result of recognition.

Do you have any solution ? or I should post the question on pull request tensorflow or ask stackoverflow

**Comment by samjabrahams** [comment]
Unfortunately, protobuf (`.pb`) files can't do any computation on their own. Think of them as being like JSON or XML- they just store information to be used by a later program. In this case the later program is TensorFlow.

You're not out of luck, yet- TensorFlow includes [instructions for building static binaries with a Makefile.](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/makefile), which is more likely to work on the Pi Zero (I can't guarantee it will work, however). You can also check out some examples [here](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/pi_examples)

**Comment by Khoa-NT** [comment]
Thank you @samjabrahams 

I already tried that tutorial.
**instructions for building static binaries with a Makefile** work great but then I stuck in the Pi_example
https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/pi_examples
because of the error `#include "tensorflow/core/framework/graph.pb.h"`

I posted the problem and still waiting  drpngx for the solution in this post:
tensorflow/tensorflow#3251 (comment)

About the protobuf:
yes, I know protobuf is a result file which contains result of weight or result of filter in neural network after training. But there should be a way that we can open protobuf and just use those result without tensorflow.
It will be great for the application of devices which cannot install tensorflow. 
Is it enough for me to pull request on tensorflow git hub ?
thank you
